ksql-functional-tests/src/test/resources/rest-query-validation-tests/insert-values.json

```diff
@@ -226,14 +226,38 @@
       "name": "should handle arbitrary expressions",
       "statements": [
         "CREATE STREAM TEST (I INT, A ARRAY<INT>) WITH (kafka_topic='test_topic', value_format='JSON');",
-        "INSERT INTO TEST (I, A) VALUES (-1, AS_ARRAY(1, 1 + 1, 3));"
+        "INSERT INTO TEST (I, A) VALUES (-1, ARRAY[1, 1 + 1, 3]);"
       ],
       "inputs": [
       ],
       "outputs": [
         {"topic": "test_topic", "key": null, "value": {"I": -1, "A": [1, 2, 3]}}
       ]
     },
+    {
+      "name": "should handle arbitrary nested expressions",
+      "statements": [
+        "CREATE STREAM TEST (I INT, A ARRAY<ARRAY<INT>>) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "INSERT INTO TEST (I, A) VALUES (-1, ARRAY[ARRAY[1]]);"
+      ],
+      "inputs": [
+      ],
+      "outputs": [
+        {"topic": "test_topic", "key": null, "value": {"I": -1, "A": [[1]]}}
+      ]
+    },
```

Comment:
I don't think we're fully testing the coercer with this test as the `1` from `ARRAY[ARRAY[1]]` will default to a `IntegerLiteral` and the type is `ARRAY<ARRAY<INT>>`.

What happens if you make it `ARRAY<ARRAY<BIGINT>>`?

Same goes for the map test below.

Reply:
done